### ISSUE TITLE
Is custom dictionary supported?

### ISSUE THREAD

**Issue by gvelasquez760** [naptha/tesseract.js#158]
Hi im trying to include my custom words in the process but isn't working, what im missing?

```
Tesseract.recognize(myImage, {
    lang: 'spa',
    load_system_dawg: true,
    load_freq_dawg: true,
	user_words_suffix: 'user-words',
	user_words_file: 'spa.user-words'
  })
```

(i have spa.user-words in the same folder)

maybe is not supported yet? or i am doing something wrong?

Thanks!

**Comment by quilicicf** [comment]
Hi, digging this one up from the grave. 

I tried to look how to add this in the list of supported parameters from `Worker#setParameters` but I'm not familiar with the code base enough to understand how to plug this. 

Any chance to get this done? I can give a hand but I'd need guidance...

**Comment by jeromewu** [comment]
Instead of digging into the code, it will be great if you can provide some tutorial about this feature and how it works in tesseract-ocr. After I totally understand how it works, I can start to work on this feature recently.

**Comment by quilicicf** [comment]
The feature is quite simple, it consists in adding words to the dictionary to help tesseract find them in the image. 

For example, if you are reading an image that you know contains people names or pseudos, you can tell it to tesseract. 

First, you create a file (here in `/tmp/user_words.txt` with a word per line like so:

```
taco778997
borilmehnir
```

Then you call tesseract like this:

```
tesseract myImage.png stdout --user-words /tmp/user_words.txt
```

**Comment by jeromewu** [comment]
Hi @quilicicf , after hours of research, I have some conclusions for your reference:

1. Right now tesseract.js is transpiled from tesseract v4, and in tesseract v4 user_words are NOT working right now. (More specifically, it is LSTM mode makes not working)
2. If you want to enforce tesseract v4 to work with user_words, you need to use legacy mode with tessedit_ocr_engine_mode equals to OEM.TESSERACT_ONLY, but it is highly not recommended as it will make the entire result of OCR worse.

In summary, it is tesseract v4 that doesn't suppport user_words, the only solution is to use legacy mode.

**Comment by quilicicf** [comment]
Damn, sorry you had to dig for that long :-( 

Thanks for doing it anyway, I hope this gets fixed in tesseract v4.

**Comment by jeromewu** [comment]
Close for now.
